test/unit/search/dream_index/interleaved_bloom_filter_test.cpp

```diff
@@ -0,0 +1,152 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/search/dream_index/interleaved_bloom_filter.hpp>
+#include <seqan3/search/dream_index/concept.hpp>
+#include <seqan3/test/cereal.hpp>
+
+template <typename ibf_type>
+struct interleaved_bloom_filter_test : public ::testing::Test
+{
+    static ibf_type make_ibf(size_t bins, size_t bits)
+    {
+        return ibf_type{seqan3::interleaved_bloom_filter{bins, bits}};
+    }
+
+    static ibf_type make_ibf(size_t bins, size_t bits, size_t num_hash)
+    {
+        return ibf_type{seqan3::interleaved_bloom_filter{bins, bits, num_hash}};
+    }
+};
+
+using ibf_types = ::testing::Types<seqan3::interleaved_bloom_filter<seqan3::data_layout::uncompressed>,
+                                   seqan3::interleaved_bloom_filter<seqan3::data_layout::compressed>>;
+
+TYPED_TEST_SUITE(interleaved_bloom_filter_test, ibf_types, );
```

Comment:
Why is there a coma after ibf_types and then nothing?

Reply:
https://github.com/google/googletest/issues/2271
Basically it's a bug in googletest and you need to put that comma there to not create a warning (=error for us).